pennylane/ops/op_math/composite.py

```diff
@@ -229,12 +229,12 @@ def eigendecomposition(self):
         """
         eigen_func = math.linalg.eigh if self.is_hermitian else math.linalg.eig
 
-        if self.hash not in self._eigs:
+        if self not in self._eigs:
```

Comment:
Maybe is just me being confused about it, but why is this not accessing the class attribute `Composite._eigs` instead of the instance attribute `self._eigs`? The former is what is done in `qml.Hermitian`

Reply:
They're the same.  Since we haven't set `self._eigs` to be anything else, we are left with the class level `_eigs`.

Reply:
Oh I see, I guess ideally we would have use it consistently in both classes, but nothing major.